### PR TITLE
updates decommisioning behavior such that an agent with a lifetime of 1 ...

### PR DESCRIPTION
**Pull request by gidden**
...built at t_0 will also be decommissioned at t_0, such that it will experience one full time step

This adds to the agent API an `exit_time` function. `-1` is returned if the agent has an 'infinite' lifetime. Otherwise, the total number of time steps - 1 is returned, per the associated [discussion](https://groups.google.com/forum/#!topic/cyclus-dev/jt29aJGkIdM).

### CODE REVIEW

**Comment by gidden** [comment]
Note that whichever PR is chosen (if any), @scopatz should definitely at least chime in, as he is able.

File: src/agent.h [hunk 1]
```diff
@@ -379,6 +379,20 @@ class Agent : public StateWrangler, virtual public Ider {
   /// decommissioning (-1 if the agent has an infinite lifetime).
   inline const int lifetime() const { return lifetime_; }
 
+  /// Returns the default time step at which this agent will exit the
+  /// simulation (-1 if the agent has an infinite lifetime).
+  //
```
Comment (rwcarlsen) [inline src/agent.h]: triple slashes all the way down.

**Comment by rwcarlsen** [comment]
Could you update the source and sink agents in cyclus to use the exit_time now too?

**Comment by rwcarlsen** [comment]
I like this better than my PR - I'll revert the commit in mine.

**Comment by gidden** [comment]
done @rwcarlsen

File: src/agent.h [hunk 2]
```diff
@@ -379,6 +379,18 @@ class Agent : public StateWrangler, virtual public Ider {
...
+  /// Returns the default time step at which this agent will exit the
+  /// simulation (-1 if the agent has an infinite lifetime).
+  ///
+  /// Deomissioning happens at the end of a time step. With a lifetime of 1, we
+  /// expect an agent to go through only 1 entire time step. In this case, the
+  /// agent should be decommissioned on the same time step it was
+  /// created. Therefore, for agents with non-infinite lifetimes, the exit_time
+  /// will be the enter time plus its lifetime less 1.
+  inline const int exit_time() const {
+    lifetime() == -1 ? return -1 : return enter_time_ + lifetime_ - 1;
```
Comment (rwcarlsen) [inline src/agent.h]: this doesn't compile.  I'd prefer

```c++
if (lifetime_ < 0) {
  return -1
}
return enter_time_+lifetime_-1;
```

Comment (rwcarlsen) [inline src/agent.h]: But at least fix compile error :-)

**Comment by gidden** [comment]
I didn't mean to leave in the ternary operator. Good catch!

**Comment by scopatz** [comment]
Please update the archetype developer's guide with this information. Thanks!

**Pull request merged**